docs/typescript.md

```diff
@@ -0,0 +1,9 @@
+This project uses [TypeScript](https://en.wikipedia.org/wiki/TypeScript).
+
+The fact that different parts of this project are built for different target architectures, namely browsers and node servers, is reflected in the TypeScript configuration. The different targets are organized in individual files. Notable differences are the `src/` files included and libraries available.
+
+* `tsconfig.base.json` The base configuration that is inherited by all other configurations
```

Comment:
Maybe we can repeat here which dir is responsible for that. (e.g. tsconfig.server hits src/server and types/server)...and so on. Just to make things more explicit...but this is totally optional.

Reply:
Tried that.